src/ansys/fluent/core/quantity.py

```diff
@@ -0,0 +1,620 @@
+# -*- coding: latin-1 -*-
+from collections import OrderedDict
+
+
+class _UnitsTable(object):
+    fundamental_units = {
+        "kg": "M",
+        "g": "M",
+        "lb": "M",
+        "lbm": "M",
+        "slug": "M",
+        "m": "L",
+        "cm": "L",
+        "ft": "L",
+        "in": "L",
+        "s": "T",
+        "A": "I",
+        "K": "?",
+        "C": "?",
+        "F": "?",
+        "R": "?",
+        "mol": "N",
+        "slugmol": "N",
+        "cd": "J",
+        "sr": "?",
+        "radian": "Angle",
+        "degree": "Angle",
+    }
+
+    derived_units = {
+        "N": "kg m s^-2",
+        "Pa": "N m^-2",
+        "W": "N m s^-1",
+        "J": "N m",
+        "V": "A ohm",
+        "F": "N m V^-2",
+        "H": "N m A^-2",
+        "S": "ohm^-1",
+        "Wb": "N m A^-1",
+        "T": "Wb m^-2",
+        "dyne": "g cm s^-2",
+        "erg": "dyne cm",
+        "pdl": "lbm ft s^-2",
+        "BTU": "J",
+        "psi": "lbf in^-2",
+        "lbf": "slug ft s^-2",
+        "psf": "lbf ft^-2",
+        "ohm": "kg m^2 s^-3 A^-2",
+        "Hz": "s^-1",
+    }
+
+    derived_units_with_conversion_factor = {
+        "l": (0.001, "m^3"),
+        "gal": (0.0037854117839999993, "m^3"),
+        "BTU": (1055.056, "J"),
+        "C": (lambda c: c + 273.15, "K"),
+        "F": (lambda f: (f - 32) * (5 / 9) + 273.15, "K"),
+    }
+
+    multipliers = {
+        "d": 10**-1,
+        "c": 10**-2,
+        "m": 10**-3,
+        "?": 10**-6,
+        "n": 10**-9,
+        "p": 10**-12,
+        "f": 10**-15,
+        "a": 10**-18,
+        "z": 10**-21,
+        "y": 10**-24,
+        "da": 10**1,
+        "h": 10**2,
+        "k": 10**3,
+        "M": 10**6,
+        "G": 10**9,
+        "T": 10**12,
+        "P": 10**15,
+        "E": 10**18,
+        "Z": 10**21,
+        "Y": 10**24,
+    }
+
+    dimension_order = OrderedDict(
+        [
+            ("Mass", "M"),
+            ("Length", "L"),
+            ("Time", "T"),
+            ("Temperature", "?"),
+            ("Angle", "Angle"),
+            ("ChemicalAmount", "N"),
+            ("Light", "J"),
+            ("Current", "I"),
+            ("SolidAngle", "?"),
+            ("", ""),
+        ]
+    )
+
+    conversion_map = {
+        "kg": 1,
+        "g": 0.001,
+        "lb": 0.45359237,
+        "lbm": 0.45359237,
+        "slug": 14.59390293720637,
+        "m": 1,
+        "cm": 0.01,
+        "ft": 0.30479999999999996,
+        "in": 0.0254,
+        "s": 1,
+        "A": 1,
+        "mol": 1,
+        "slugmol": 14593.9,
+        "cd": 1,
+        "sr": 1,
+        "radian": 1,
+        "degree": 0.017453292519943295,
+        "K": 1,
+        "C": 1,
+        "F": 0.5555555555555556,
+        "R": 0.5555555555555556,
+    }
```

Comment:
As discussed with @hpohekar we don't need to see any conversion data defined elsewhere, especially not inside functions. We can define factor and optional offset in each case, so temperature (and anything else where relative scales are used) can live together with everything else.

Reply:
Now we are supporting all temperature conversions.  We will definitely improve this one. Thank you.